models/model.js

```diff
@@ -22,15 +22,28 @@ exports.selectArticle = (article_id) => {
 }
 
 exports.selectAllArticles = () => {
-
-
     return connection
     .query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count
     FROM articles
     LEFT JOIN comments ON articles.article_id = comments.article_id
     GROUP BY articles.article_id
     ORDER BY articles.created_at DESC;`)
     .then(({rows}) => {
+        
+        return rows
+    })
+}
+
+exports.selectCommentsByArticleId = (article_id) => {
+    return connection
+    .query(`SELECT *
+    FROM comments
+    WHERE article_id = $1
+    ORDER BY created_at DESC`, [article_id])
+    .then(({rows}) => {
+        if(rows.length === 0) {
+            return Promise.reject({status: 404, msg: 'Not Found'})
+        }
```

Comment:
this logic won't work here for this use case.
If the array is returned but it's empty that just means although the article might exist there is just no comments for it which is still a valid request.